src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperRequiredAttributeDescriptor.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Compilation.TagHelpers
+{
+    public class TagHelperRequiredAttributeDescriptor
+    {
+        public string Name { get; set; }
+
+        public string Value { get; set; }
+
+        public char Selector { get; set; }
+
+        public bool IsCSSSelector { get; set; }
+
+        public bool Matches(string attributeName, string attributeValue)
+        {
+            if (IsCSSSelector)
+            {
+                var nameMatches = string.Equals(Name, attributeName, StringComparison.OrdinalIgnoreCase);
+
+                if (!nameMatches)
+                {
+                    return false;
+                }
+
+                var valueMatches = false;
+
+                switch (Selector)
+                {
+                    case '^': // Value starts with
+                        valueMatches = attributeValue.StartsWith(Value, StringComparison.Ordinal);
+                        break;
+                    case '$': // Value ends with
+                        valueMatches = attributeValue.EndsWith(Value, StringComparison.Ordinal);
+                        break;
+                    case '=': // Value equals
+                        valueMatches = string.Equals(attributeValue, Value, StringComparison.Ordinal);
+                        break;
+                    default: // No value selector, force true because at least the attribute name matched.
+                        valueMatches = true;
+                        break;
+                }
+
+                return valueMatches;
+            }
+            else if (Selector == '*')
+            {
+                return attributeName.Length != Name.Length &
+                    attributeName.StartsWith(Name, StringComparison.OrdinalIgnoreCase);
+            }
+            else
+            {
+                return string.Equals(Name, attributeName, StringComparison.OrdinalIgnoreCase);
+            }
+        }
+
+        public static bool SupportsSelector(char selector, bool isCSSSelector)
```

Comment:
`IsSupportedCSSOperator()` and remove `isCSSSelector` parameter and treatment of `'*'` as a "selector".